urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/Game_World/GameWorldLevelBuilder.java

```diff
@@ -0,0 +1,177 @@
+package Game_World;
+
+import java.awt.Point;
+import java.beans.IntrospectionException;
+import java.io.FileNotFoundException;
+
+import Data_Storage.XMLWriter;
+import Game_World_Objects.*;
+
+/**
+ * The Class represents the completed levels of the GameWorld, it uses the base templates from the BaseLocations
+ * and the GameObjects in the GameWorld Objects package to create a whole level.
+ * This class currently only creates level one so far.
+ * @author JoshBrake
+ * @author hoschurayn
+ * @author Danesh Abeyratne
+ *
+ */
+public class GameWorldLevelBuilder {
+
+	private GameWorld level;
+	private String levelName = null;
+
+	public GameWorldLevelBuilder(String levelName) {
+		this.levelName = levelName;
+		createLevel();
+	}
+
+	private void createLevel() {
+		if (levelName.equals("One")) {
+			level = getLevelOne();
+			writeXMLLevel();
+		}
+	}
+	/**
+	 * Returns level one of the Game
+	 * @return GameWorld
+	 *
+	 */
+	public static GameWorld getLevelOne() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 64 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 87-93 into a separate method.